src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -117721,6 +117721,142 @@ static T F9()
                 Diagnostic(ErrorCode.WRN_NullReferenceReturn, "t7").WithLocation(51, 16));
         }
 
+        [Fact]
+        [WorkItem(38638, "https://github.com/dotnet/roslyn/issues/38638")]
+        public void MaybeNullT_Meet_02()
+        {
+            var source =
+@"#nullable enable
+class C<T> where T : new()
+{
+    static void F0(T t)
+    {
+    }
+    static T F2(T t)
+    {
+        T t2 = t;
+        T r2;
+        try
+        {
+            t2 = default(T);
+            t2 = t;
+        }
+        finally
+        {
+            F0(t2); // 1
```

Comment:
I'm a little confused by this-- is it expected that we could throw out of the 'try' block at any point and then run the 'finally' block, so we have to be as pessimistic as possible here?

Reply:
Yes, we assume an exception could have been thrown between the assignments above, although perhaps we could do better for those simple assignments.

---
In reply to: [348225225](https://github.com/dotnet/roslyn/pull/39778#discussion_r348225225) [](ancestors = 348225225)